analyticsservice/analytics.go

```diff
@@ -43,13 +45,17 @@ type analyticsEvent struct {
 
 // AnalyticsService provides a way to cache and send analytics to the server
 type AnalyticsService struct {
-	mx            *sync.Mutex
-	analyticsChan chan analyticsEvent
-	analyticsData map[string]analyticsEvent
-	timeout       time.Duration
-	logger        logger.Logger
-	metricsClient *metricsclient.ClientWithResponsesInterface
-	environmentID string
+	analyticsChan          chan analyticsEvent
+	evaluationsAnalyticsMx *sync.Mutex
+	targetAnalyticsMx      *sync.Mutex
+	seenTargetsMx          *sync.RWMutex
+	evaluationAnalytics    map[string]analyticsEvent
+	targetAnalytics        map[string]evaluation.Target
+	seenTargets            map[string]bool
+	timeout                time.Duration
+	logger                 logger.Logger
+	metricsClient          metricsclient.ClientWithResponsesInterface
```

Comment:
This was a pointer to an interface which I removed as it prevented me from using my own types for testing purposes.